perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/SearcherBlImpl.java

```diff
@@ -84,6 +89,38 @@ public SearcherBlImpl(SearcherImplApi searcherImpl) {
 		return this.getUsersForCoreAttributesByMapOfAttributes(sess, mapOfCoreAttributesWithValues);
 	}
 
+	@Override
+	public List<Group> getGroups(PerunSession sess, Vo vo, Map<String, String> attributesWithSearchingValues) throws AttributeNotExistsException {
+		return this.getGroups(sess, attributesWithSearchingValues).stream()
```

Comment:
Up to discussion (performance): Both methods getGroups should firstly retrieve all necessary groups. Meaning, that this method retrieves all Vo groups and the second one retrieves all groups. Then they could both use some private method which takes these groups as a parameter alongside the map of attributes. According to these parameters, the private method returns only sufficient groups.

Reply:
You are right, there is a way to do it more efficiently, but we would need to change methods in the implementation layer adequately. When we found this method is too slow, we can change it later.